src/common/Cryptography/SRP6.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef TRINITY_SRP6_H
+#define TRINITY_SRP6_H
+
+#include "BigNumber.h"
+#include "Define.h"
+#include "Common.h"
+#include "CryptoHash.h"
+#include <array>
+#include <optional>
+
+namespace Trinity::Crypto
+{
+    class TC_COMMON_API SRP6
+    {
+        public:
+            static constexpr size_t SEED_LENGTH = 32;
+            using Seed = std::array<uint8, SEED_LENGTH>;
+            static constexpr size_t VERIFIER_LENGTH = 32;
+            using Verifier = std::array<uint8, VERIFIER_LENGTH>;
+            static constexpr size_t EPHEMERAL_KEY_LENGTH = 32;
+            using EphemeralKey = std::array<uint8, EPHEMERAL_KEY_LENGTH>;
+
+            static const std::array<uint8, 1> g;
+            static const std::array<uint8, 32> N;
+
+            // this is the old sha_pass_hash hack
+            [[deprecated("No. Bad. You shouldn't even HAVE that. Stop right now and use MakeRegistrationData instead.")]]
```

Comment:
I'm moderately fine with quips in commit messages but please avoid in strings possibly shown by the compiler.

Reply:
I hate them in commit messages, in compiler output and comments. Remove ALL OF THAT